### PR TITLE
remove correct file if failed to get root_authorized_keys

### PR DESCRIPTION
**Pull request by sjorge**

### CODE REVIEW

File: src/lib/smartdc/mdata-fetch [hunk 1]
```diff
@@ -85,7 +85,7 @@ if [[ ! -f /root/.ssh/authorized_keys ]]; then
       ;;
     1)
       smartdc_info "root_authorized_keys not defined."
-      rm -f /var/svc/mdata-operator-script{,.new}
+      mv /root/.ssh/authorized_keys.new
```
Comment (chorrell) [inline src/lib/smartdc/mdata-fetch]: This should actually be:

```
rm -f /root/.ssh/authorized_keys{,.new}
```

**Comment by sjorge** [comment]
Yeah that looks correct, disregards the comment in my previous e-mail too. I skipped over the check earlier on.

**Comment by chorrell** [comment]
Great, can you make that change and push another commit?

**Comment by sjorge** [comment]
Done. For completeness, my remark was that this would clobber /root/.ssh/authorized_keys if we remove both. This is not the case because there is a check beforehand to skip this if the file exists.

**Comment by chorrell** [comment]
LGTM

**Pull request merged**